ledger/src/blockstore.rs

```diff
@@ -1483,21 +1507,25 @@ impl Blockstore {
             let leader_pubkey = leader_schedule
```

Comment:
I see now... Here is the logic where we end up passing in shreds where index > last. This index could also be a last or not.

One thing that's a little odd is that I believe we could call a slot duplicate just for screwing up the last flag (there don't need to actually be duplicate slots/shreds). It's probably fine, but interesting nonetheless

Reply:
that's true, we would prematurely call such a slot duplicate. I think it's fine because this slot should be unable to replay due to the invalid flag anyway.